src/org/opensolaris/opengrok/configuration/Configuration.java

```diff
@@ -572,12 +572,12 @@ public void setHandleHistoryOfRenamedFiles(boolean enable) {
         this.handleHistoryOfRenamedFiles = enable;
     }
 
-    public List<Project> getProjects() {
+    public Map<String,Project> getProjects() {
         return projects;
     }
 
-    public void setProjects(List<Project> projects) {
-        this.projects = projects;
+    public void setProjects(Map<String,Project> projects) {
+        this.projects = (HashMap<String, Project>) projects;
```

Comment:
if you do it [here](https://github.com/OpenGrok/OpenGrok/pull/1543/files#r113946427) the this cast is not necessary